stories/nav-drawer.stories.ts

```diff
@@ -0,0 +1,169 @@
+  component: 'igc-nav-drawer',
+  parameters: {
+    actions: {
+      handles: ['igcOpening', 'igcOpened', 'igcClosing', 'igcClosed'],
+    },
+  },
+  argTypes: {
+    open: {
+      type: 'boolean',
+      description: 'Determines whether the drawer is opened.',
+      control: 'boolean',
+      table: {
+        defaultValue: {
+          summary: false,
+        },
+      },
+    },
+    pinned: {
+      type: 'boolean',
+      description: 'Determines whether the drawer is pinned.',
+      control: 'boolean',
+      table: {
+        defaultValue: {
+          summary: false,
+        },
+      },
+    },
+    disabled: {
+      type: 'boolean',
+      description: 'Determines whether a drawer item is disabled.',
+      control: 'boolean',
+      table: {
+        defaultValue: {
+          summary: false,
+        },
+      },
+    },
+    active: {
+      type: 'boolean',
+      description: 'Determines whether a drawer item is active.',
+      control: 'boolean',
+      table: {
+        defaultValue: {
+          summary: false,
+        },
+      },
+    },
+    position: {
+      type: '"start" | "end" | "top" | "bottom"',
+      options: ['start', 'end', 'top', 'bottom'],
+      control: {
+        type: 'inline-radio',
+      },
+      table: {
+        defaultValue: {
+          summary: 'start',
+        },
+      },
+    },
+  },
+};
+
+interface ArgTypes {
+  content1: string;
+  content2: string;
+  open: boolean;
+  pinned: boolean;
+  disabled: boolean;
+  active: boolean;
+  position: 'start' | 'end' | 'top' | 'bottom';
+}
+
+//end region
+
+registerIcon(
+  'home',
+  'https://unpkg.com/material-design-icons@3.0.1/action/svg/production/ic_home_24px.svg'
+);
+
+registerIcon(
+  'search',
+  'https://unpkg.com/material-design-icons@3.0.1/action/svg/production/ic_search_24px.svg'
+);
+
+const handleOpen = () => {
+  const drawer = document.querySelector(
+    'igc-nav-drawer'
+  ) as IgcNavDrawerComponent;
+  drawer?.show();
+};
+
+const handleClose = () => {
+  const drawer = document.querySelector(
+    'igc-nav-drawer'
+  ) as IgcNavDrawerComponent;
+  drawer?.hide();
+};
+
+const handleToggle = () => {
+  const drawer = document.querySelector(
+    'igc-nav-drawer'
+  ) as IgcNavDrawerComponent;
+  drawer?.toggle();
+};
+
+const Template: Story<ArgTypes, Context> = (
+  {
+    content1 = 'Home',
+    content2 = 'Search',
+    open = false,
+    pinned = false,
+    disabled = false,
+    active = false,
+    position,
+  }: ArgTypes,
+  { globals: { direction } }: Context
+) => {
+  return html`
+    <div style="display: flex;">
+      <igc-nav-drawer
+        dir=${ifDefined(direction)}
+        .open=${open}
+        .pinned=${pinned}
+        .position=${position}
+      >
+        <igc-nav-drawer-header-item>Sample Drawer</igc-nav-drawer-header-item>
+
+        <igc-nav-drawer-item .disabled=${disabled}>
+          <igc-icon slot="icon" name="home"></igc-icon>
+          <span slot="text">${content1}</span>
```

Comment:
Update this template - it seems like slot="text" does not exist on the drawer item component.

Reply:
We needed a name for the slot in order to style the items properly.